### DISCUSSION TITLE
Advanced templates

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#3569] [Q&A] [answered]
Is it possible to have html templates pass variables to blocks?

I mean have `for` loop and `if` statement inside that would say that if `X` then place `Y` block/slot here.

I am thinking would it be possible to use PB as cms with webpage block builder by just using JS extending.

**Comment by ganigeorgiev** [comment, marked as answer]
You should be able to wrap blocks with `{{if condition}}...{{end}}` to conditionally render them if that is what you are looking for. You can also pass any variable to the block by specifying its 3rd argument (aka. `{{block "example" yourVar}}...{{end}}`) or if you want to pass all currently scoped data to the block definition you can use just "dot" (`.`).

For more details about the Go templates syntax you can check the HashiCorp tutorial at https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax.

**Reply by pietru2004** [reply]
ok thanks
